core/templates/pages/topic-viewer-page/story-node-tile/story-node-tile.component.spec.ts

```diff
@@ -0,0 +1,523 @@
+    spyOn(urlService, 'addField').and.callThrough();
+    expect(componentInstance.getStartLink()).toEqual(
+      '/explore/' +
+        componentInstance.explorationId +
+        '?story_url_fragment=' +
+        componentInstance.storyUrlFragment +
+        '&topic_url_fragment=' +
+        componentInstance.topicUrlFragment +
+        '&classroom_url_fragment=' +
+        componentInstance.classroomUrlFragment +
+        '&node_id=' +
+        mockStoryNode.getId()
+    );
+  });
+
+  it('should generate correct path icon parameters', () => {
+    componentInstance.storyId = 'story_1';
+    componentInstance.thumbnailFilename = 'thumbnail.png';
+    componentInstance.thumbnailBgColor = '#FFFFFF';
+    componentInstance.generatePathIconParameters();
+    componentInstance.node = mockStoryNode;
+    componentInstance.ngOnInit();
+    expect(componentInstance.generatePathIconParameters()).toEqual([
+      {
+        thumbnailIconUrl: 'thumbnail_url',
+        left: '225px',
+        top: '35px',
+        thumbnailBgColor: '#FFFFFF',
+      },
+    ]);
+  });
+  it('should call getCheckpointCount and getCompletedProgressStatus', waitForAsync(() => {
+    spyOn(componentInstance, 'getCheckpointCount').and.callThrough();
+    spyOn(
+      readOnlyExplorationBackendApiService,
+      'fetchExplorationAsync'
+    ).and.returnValue(Promise.resolve(sampleExpResponse));
+    componentInstance.getCheckpointCount();
+    componentInstance.node = mockStoryNode;
+    componentInstance.ngOnInit();
+    expect(componentInstance.checkpointCount).toEqual(3);
+    expect(componentInstance.totalCheckpointLoaded).toEqual(true);
+    spyOn(componentInstance, 'getCompletedProgressStatus').and.callThrough();
+  }));
+
+  it(
+    'should get status of checkpoints weather it is completed,in-progress,' +
+      'and non-completed',
+    () => {
+      componentInstance.nodeDataLoaded = true;
+      componentInstance.totalCheckpointLoaded = true;
+      componentInstance.checkpointCount = 3;
+      componentInstance.completedcheckpoints = 1;
+      componentInstance.checkpointStatusArray = [
+        'completed',
+        'in-progress',
+        'incomplete',
+      ];
+      componentInstance.getCompletedProgressStatus();
+      componentInstance.node = mockStoryNode;
+      componentInstance.ngOnInit();
+      expect(componentInstance.checkpointStatusArray).toEqual([
+        'completed',
+        'in-progress',
+        'incomplete',
+      ]);
+    }
+  );
+
+  it('should get the completed progress bar width', () => {
+    componentInstance.checkpointCount = 3;
+    componentInstance.completedcheckpoints = 0;
+    expect(componentInstance.getCompletedProgressBarWidth()).toEqual(0);
+    componentInstance.completedcheckpoints = 1;
+    expect(componentInstance.getCompletedProgressBarWidth()).toEqual(25);
+    componentInstance.completedcheckpoints = 2;
+    expect(componentInstance.getCompletedProgressBarWidth()).toEqual(75);
```

Comment:
Help me to understand the math here, if two checkpoints are completed then how come the progress is 75?